tdrs-backend/tdpservice/parsers/schema_defs/ssp/m5.py

```diff
@@ -0,0 +1,138 @@
+                  ),
+              validators.if_then_validator(
+                        condition_field='FAMILY_AFFILIATION', condition_function=validators.isInLimits(1, 2),
+                        result_field='PARENT_MINOR_CHILD', result_function=validators.isInLimits(1, 3),
+                  ),
+              validators.if_then_validator(
+                        condition_field='FAMILY_AFFILIATION', condition_function=validators.isInLimits(1, 3),
+                        result_field='EDUCATION_LEVEL', result_function=validators.or_validators(
+                            validators.isInStringRange(1, 16), validators.isInStringRange(98, 99)),
+                  ),
+              validators.if_then_validator(
+                        condition_field='FAMILY_AFFILIATION', condition_function=validators.matches(1),
+                        result_field='CITIZENSHIP_STATUS', result_function=validators.isInLimits(1, 3),
+                  ),
+              validators.if_then_validator(
+                        condition_field='DATE_OF_BIRTH', condition_function=validators.olderThan(18),
+                        result_field='REC_OASDI_INSURANCE', result_function=validators.isInLimits(1, 2),
+                  ),
+              validators.if_then_validator(
+                        condition_field='FAMILY_AFFILIATION', condition_function=validators.matches(1),
+                        result_field='REC_FEDERAL_DISABILITY', result_function=validators.isInLimits(1, 2),
+                  ),
+          ],
+          fields=[
+              Field(item="0", name='RecordType', type='string', startIndex=0, endIndex=2,
+                    required=True, validators=[]),
+              Field(item="3", name='RPT_MONTH_YEAR', type='number', startIndex=2, endIndex=8,
+                    required=True, validators=[validators.dateYearIsLargerThan(1998),
+                                               validators.dateMonthIsValid(),]),
+              Field(item="5", name='CASE_NUMBER', type='string', startIndex=8, endIndex=19,
+                    required=True, validators=[validators.isAlphaNumeric()]),
+              Field(item="13", name='FAMILY_AFFILIATION', type='number', startIndex=19, endIndex=20,
+                    required=True, validators=[]),
+              Field(item="14", name='DATE_OF_BIRTH', type='string', startIndex=20, endIndex=28,
+                    required=True, validators=[validators.dateYearIsLargerThan(1998),
+                                               validators.dateMonthIsValid(),]),
+              TransformField(transform_func=ssp_ssn_decryption_func, item="15", name='SSN', type='string',
+                             startIndex=28, endIndex=37, required=True, validators=[validators.validateSSN()],
+                             is_encrypted=False),
+              Field(item="16A", name='RACE_HISPANIC', type='number', startIndex=37, endIndex=38, required=True,
+                    validators=[validators.validateRace()]),
```

Comment:
this is new to me but the m4/m5 spreadsheet has the race fields accepting `0-2, " "` which makes me think we should have `required=False` - we'd have to change that in all the schemas though

Reply:
this applies for many of the fields below including `REC_FEDERAL_DISABILITY`, `REC_AID_TOTALLY_DISABLED`, etc

Reply:
Right, I didn't handle the blank values yet because I am waiting for the 2695 PR to merge. That way I didnt have to implement things twice. I could preemptively set `required=False` for M4 and M5.